platform/core/src/services/ToolBarService/ToolbarService.ts

```diff
@@ -390,16 +398,38 @@ export default class ToolbarService extends PubSubService {
    * @param {string} key - The key of the button section.
    * @param {Array} buttons - The buttons to be added to the section.
    */
-  createButtonSection(key, buttons) {
-    if (this.state.buttonSections[key]) {
-      this.state.buttonSections[key].push(
-        ...buttons.filter(
-          button => !this.state.buttonSections[key].find(sectionButton => sectionButton === button)
-        )
-      );
-    } else {
+  createButtonSection(key: string, buttons: string[]) {
+    // Create or update the button section
+    if (!this.state.buttonSections[key]) {
       this.state.buttonSections[key] = buttons;
+    } else {
```

Comment:
I don't think you should have to know anything about the nested button section - that update should be handled by an entirely different useToolbar, and should get the appropriate enhanced information when the toolbar is actually used.